MantidQt/SliceViewer/src/SliceViewer.cpp

```diff
@@ -65,6 +65,18 @@ using Poco::XML::NodeIterator;
 using Poco::XML::NodeFilter;
 using MantidQt::API::AlgorithmRunner;
 
+namespace {
+/*
+Checks if the rebinning is in a consistent state, ie if rebin mode is selected and there
+is a rebin workspace or there is no rebin workspace and no rebin mode selected.
+The state is inconsistent if rebin mode is selected and there is no workspace.
+*/
+  bool isRebinInConsistentState(Mantid::API::IMDWorkspace_sptr rebinnedWS, bool useRebinMode) {
```

Comment:
My only comment here (not worth changing) is that passing the shared_ptr implies additional, or change of ownership and that's not what's happening here. Would be better to pass `Mantid::API::IMDWorkspace_sptr&`. Furthermore, use the `.get` accessor and compare to null_ptr might be have made this more explicit as to what's being tested.